src/args.rs

```diff
@@ -76,36 +76,47 @@ fn parse_interface(src: &str) -> Result<IpAddr, std::net::AddrParseError> {
 }
 
 /// Checks wether the auth string is valid, i.e. it follows the syntax username:password
-fn parse_auth(src: &str) -> Result<(String, String), String> {
-    let mut split = src.splitn(2, ':');
+fn parse_auth(src: &str) -> Result<auth::RequiredAuth, String> {
+    let mut split = src.splitn(3, ':');
+    let errmsg = "Invalid credentials string, expected format is username:password".to_owned();
 
     let username = match split.next() {
         Some(username) => username,
-        None => {
-            return Err(
-                "Invalid credentials string, expected format is username:password".to_owned(),
-            )
-        }
+        None => return Err(errmsg),
     };
 
-    let password = match split.next() {
+    let second_part = match split.next() {
```

Comment:
Please document the semantics of `second_part`. I know it's either just plain password or hash stuff but this doesn't become obvious from the new name.

Reply:
What do you prefer? A comment or a long but descriptive variable name?

Reply:
I think a comment is a more suitable measure here.